src/Payum/Core/Request/RenderTemplate.php

```diff
@@ -60,4 +60,13 @@ public function setResult($result)
     {
         $this->result = $result;
     }
+
+    /**
+     * @param string $name
+     * @param mixed  $value
+     */
+    public function addParameter($name, $value)
```

Comment:
let's add one more merhod `setParamter`. and this one must throw exception if there is already such parameter set where `setParameter` should always set\overwrite it. 

Could you rename context to `parameters` and gettter too.